experimental/origin-graphql/test/index.js

```diff
@@ -45,54 +46,250 @@ describe('Marketplace', function() {
     assert(Marketplace)
   })
 
-  it('should allow a listing to be created', async function() {
-    const events = await mutate(mutations.CreateListing, {
-      deposit: '0',
-      depositManager: Arbitrator,
-      from: Seller,
-      data: {
-        title: 'Test Listing',
-        description: 'Test description',
-        price: {
-          currency: ZeroAddress,
-          amount: '0.01'
-        },
-        category: 'Test category',
-        subCategory: 'Test sub-category',
-        unitsTotal: 1
-      }
-    }, true)
-    assert(events.ListingCreated)
+  it('should allow an affiliate to be added', async function() {
+    await mutate(mutations.AddAffiliate, {
+      from: Admin,
+      affiliate: Affiliate
+    })
   })
 
-  it('should allow an offer to be created', async function() {
-    const events = await mutate(mutations.MakeOffer, {
-      listingID: '999-0-0',
-      from: Buyer,
-      finalizes: 123,
-      affiliate: ZeroAddress,
-      commission: '0',
-      value: '0.1',
-      currency: ZeroAddress,
-      arbitrator: Arbitrator,
-      quantity: 1
-    }, true)
-    assert(events.OfferCreated)
-  })
+  describe('Single-unit listing', function() {
+    it('should allow a listing to be created', async function() {
+      const events = await mutate(mutations.CreateListing, {
+        deposit: '0',
+        depositManager: Arbitrator,
+        from: Seller,
+        data: {
+          title: 'Test Listing',
+          description: 'Test description',
+          price: {
+            currency: ZeroAddress,
+            amount: '0.01'
+          },
+          category: 'Test category',
```

Comment:
yes, we should do that in another PR. I think we should put a concerted effort into validating the data that goes in, because that will help catch various dapp2 bugs early